x-pack/plugins/asset_manager/server/lib/accessors/hosts/get_hosts_by_signals.ts

```diff
@@ -12,13 +12,13 @@ import { collectHosts } from '../../implicit_collection/collectors/hosts';
 export async function getHostsBySignals(
   options: GetHostsOptionsInjected
 ): Promise<{ hosts: Asset[] }> {
-  const hosts = await collectHosts({
+  const { assets } = await collectHosts({
     client: options.esClient,
     from: options.from,
-    // TODO: implement "to" for collectHosts
+    to: options.to,
     sourceIndices: options.sourceIndices,
   });
   return {
-    hosts,
+    hosts: assets,
```

Comment:
To get pagination collapse queries now sort results by the same field we collapse on. Instead of returning the `n` most recent results we'll now get the `n` first hosts by hostname when querying signals. Baring the limited sorting flexibility this sounds like a sane contract for an API if we send back the page cursor so that consumer can loop through, but we may send incomplete results when consumer rely on sliding time ranges (ie last 15mn)
We could also have the API accumulating pages in a single call and return the entire dataset but are there use cases for that